test/unit/blocks_sensing.js

```diff
@@ -229,6 +229,33 @@ test('loud? boolean', t => {
     t.end();
 });
 
+test('get attribute of sprite variable', t => {
+    const rt = new Runtime();
+    const sensing = new Sensing(rt);
+    const s = new Sprite();
+    const target = new RenderedTarget(s, rt);
+    const thing = {
+        name: 'cars',
+        value: 'trucks'
+    };
+    rt.getSpriteTargetByName = () => target;
+    target.lookupVariableByNameAndType = () => thing;
+    t.equal(sensing.getAttributeOf({PROPERTY: 'cars'}), 'trucks');
+
+    t.end();
```

Comment:
I think these tests would be more useful if you mocked out the actual variable and added it to the target (and then called getAttributeOf with both the property and the object set, what do you think?

Reply:
I'm torn. IIUC, I think doing that means testing the functionality of lookupVariableByNameAndType, which should (at least theoretically) be tested somewhere else.  Maybe it isn't though?  To me, it feels less like a unit test if I do it the way you suggested, but probably would increase our test coverage numbers.

Reply:
Part of what we're testing is whether you're providing things like `lookupVariableByNameAndType` the right arguments inside of `getAttributeOf`. Stubbing out `lookupVariableByNameAndType` doesn't get at that and thus isn't really testing whether the logic inside of `getAttributeOf` is correct. I don't think changing it to remove the `lookupVariableByNameAndType` stub would make this a duplicate test of the `lookupVariableByNameAndType` functionality but rather a test of whether the `getAttributeOf` logic functions correctly as a whole...

IMO, it's still a unit test, just testing `getAttributeOf` as one whole unit. Even if we change the internals of `getAttributeOf` and maybe even stop using `lookupVariableByNameAndType`, I would still expect its unit test to work assuming we're not changing the fundamental functionality of what `getAttributeOf` does.

Thoughts?

Reply:
Now I'm a little torn too haha.

Reply:
Went back to not mocking lookupVariableByNameAndType. I couldn't add variables with target.createVariable because I couldn't figure out how to have it set a value too so I just smashed it right into the dictionary... But, maybe I'm just missing something.